cetmix_tower_server/models/cx_tower_file_template.py

```diff
@@ -136,45 +136,71 @@ def create_file(self, server, server_dir="", raise_if_exists=False):
         :param server: recordset
             The server (cx.tower.server) on which the file should be created. This is a
             required parameter.
+        :param if_file_exists: str, optional
+            Defines the behavior if the file already exists on the server.
         :param server_dir: str, optional
             The directory on the server where the file should be created. If not set,
             the server_dir field of the template will be used.
-        :param raise_if_exists: bool, optional
-            If set to True, raises a ValidationError if the file already exists on the
-            server. Defaults to False.
 
         :return: cx.tower.file
             Returns the newly created file record (cx.tower.file) if the file was
-            created successfully. Returns the existing file record if the file already
-            exists and raise_if_exists is False.
+            created successfully or if_file_exists is set to "overwrite".
+            Returns the existing file record if the file already exists
+            and if_file_exists is set to "skip".
 
         :raises ValidationError:
-            If the file already exists on the server and raise_if_exists is True.
+            If the file already exists on the server if_file_exists is set to "raise".
         """
         self.ensure_one()
+        # Explicit guard against invalid behavior values
+        valid_behaviors = {"skip", "raise", "overwrite"}
+        if if_file_exists not in valid_behaviors:
+            raise ValidationError(
+                f"Invalid if_file_exists value: {if_file_exists}. "
+                f"Expected one of {valid_behaviors}."
+            )
         file_model = self.env["cx.tower.file"]
-        existing_file = file_model.search(
+        existing_file_ids = file_model.search(
             [
-                ("template_id", "=", self.id),
                 ("server_id", "=", server.id),
                 ("server_dir", "=", server_dir or self.server_dir),
                 ("source", "=", self.source),
             ],
-            limit=1,
+            order="id DESC",
         )
-        if existing_file:
-            if raise_if_exists:
-                raise ValidationError(_("File already exists on server."))
+        # Filter existing files by name or rendered name
+        existing_file_ids = existing_file_ids.filtered(
+            lambda f: f.name == self.file_name or f.rendered_name == self.file_name
```

Comment:
You should filter by the [full server path](https://github.com/cetmix/cetmix-tower/blob/d1fa7bb2388523bad944ac9679de1bc255b91dd1/cetmix_tower_server/models/cx_tower_file.py#L66) because it contains both render file name and directory.